### PR TITLE
Drop some babel config

### PR DESCRIPTION
**Pull request by fregante**
It seems to me that using babel in this project goes back quite a bit, and it was just carried over for inertia. Is that correct?

Why is there babel-env? Does this package run in the browser? I ran some tests and here are the results

## Current config

```is
asset addons-linter.js 727 KiB [compared for emit] (name: addons-linter) 1 related asset
asset rules/javascript/webextension-deprecated-api.js 511 KiB [compared for emit] (name: rules/javascript/webextension-deprecated-api) 1 related asset
asset rules/javascript/webextension-unsupported-api.js 469 KiB [compared for emit] (name: rules/javascript/webextension-unsupported-api) 1 related asset
asset rules/javascript/webextension-api-compat-android.js 469 KiB [compared for emit] (name: rules/javascript/webextension-api-compat-android) 1 related asset
asset rules/javascript/webextension-api-compat.js 469 KiB [compared for emit] (name: rules/javascript/webextension-api-compat) 1 related asset
asset rules/javascript/webextension-api.js 458 KiB [compared for emit] (name: rules/javascript/webextension-api) 1 related asset
asset rules/javascript/deprecated-entities.js 85.3 KiB [compared for emit] (name: rules/javascript/deprecated-entities) 1 related asset
asset rules/javascript/global-require-arg.js 84.9 KiB [compared for emit] (name: rules/javascript/global-require-arg) 1 related asset
asset rules/javascript/opendialog-remote-uri.js 84.7 KiB [compared for emit] (name: rules/javascript/opendialog-remote-uri) 1 related asset
asset rules/javascript/opendialog-nonlit-uri.js 84.7 KiB [compared for emit] (name: rules/javascript/opendialog-nonlit-uri) 1 related asset
asset rules/javascript/content-scripts-file-absent.js 45.3 KiB [compared for emit] (name: rules/javascript/content-scripts-file-absent) 1 related asset
asset rules/javascript/event-listener-fourth.js 43.5 KiB [compared for emit] (name: rules/javascript/event-listener-fourth) 1 related asset
runtime modules 793 bytes 4 modules
cacheable modules 735 KiB
  modules by path ./src/schema/ 450 KiB 64 modules
  modules by path ./src/rules/ 22.3 KiB 16 modules
  modules by path ./src/*.js 95.5 KiB 11 modules
  modules by path ./src/messages/*.js 52.5 KiB 9 modules
  modules by path ./src/scanners/*.js 50.8 KiB 8 modules
  modules by path ./src/parsers/*.js 60.3 KiB 6 modules
  ./package.json 3.71 KiB [built] [code generated]
  ./src/blocked_content_script_hosts.txt 45 bytes [built] [code generated]
modules by path external "@babel/runtime/ 672 bytes 16 modules
modules by path external "addons-scanner-utils/dist/ 84 bytes
  external "addons-scanner-utils/dist/io/utils" 42 bytes [built] [code generated]
  external "addons-scanner-utils/dist/io" 42 bytes [built] [code generated]
27 modules
webpack 5.24.3 compiled successfully in 4758 ms
```

<details>
<summary>Outdated details</summary>

## Node browserlist

With just this configuration, size and build time drop considerably. 

```json
"browserslist": ["node >=10.13"]
```

```js
asset addons-linter.js 641 KiB [emitted] (name: addons-linter) 1 related asset
asset rules/javascript/webextension-deprecated-api.js 497 KiB [emitted] (name: rules/javascript/webextension-deprecated-api) 1 related asset
asset rules/javascript/webextension-unsupported-api.js 460 KiB [emitted] (name: rules/javascript/webextension-unsupported-api) 1 related asset
asset rules/javascript/webextension-api-compat-android.js 459 KiB [emitted] (name: rules/javascript/webextension-api-compat-android) 1 related asset
asset rules/javascript/webextension-api-compat.js 459 KiB [emitted] (name: rules/javascript/webextension-api-compat) 1 related asset
asset rules/javascript/webextension-api.js 450 KiB [emitted] (name: rules/javascript/webextension-api) 1 related asset
asset rules/javascript/deprecated-entities.js 74.2 KiB [emitted] (name: rules/javascript/deprecated-entities) 1 related asset
asset rules/javascript/global-require-arg.js 73.9 KiB [emitted] (name: rules/javascript/global-require-arg) 1 related asset
asset rules/javascript/opendialog-remote-uri.js 73.7 KiB [emitted] (name: rules/javascript/opendialog-remote-uri) 1 related asset
asset rules/javascript/opendialog-nonlit-uri.js 73.6 KiB [emitted] (name: rules/javascript/opendialog-nonlit-uri) 1 related asset
asset rules/javascript/content-scripts-file-absent.js 39.2 KiB [emitted] (name: rules/javascript/content-scripts-file-absent) 1 related asset
asset rules/javascript/event-listener-fourth.js 37.5 KiB [emitted] (name: rules/javascript/event-listener-fourth) 1 related asset
runtime modules 793 bytes 4 modules
modules by path ./src/ 640 KiB 115 modules
modules by path external "@babel/runtime/helpers/ 168 bytes
  external "@babel/runtime/helpers/interopRequireWildcard" 42 bytes [built] [code generated]
  external "@babel/runtime/helpers/interopRequireDefault" 42 bytes [built] [code generated]
  external "@babel/runtime/helpers/asyncToGenerator" 42 bytes [built] [code generated]
  external "@babel/runtime/helpers/defineProperty" 42 bytes [built] [code generated]
modules by path external "addons-scanner-utils/dist/ 84 bytes
  external "addons-scanner-utils/dist/io/utils" 42 bytes [built] [code generated]
  external "addons-scanner-utils/dist/io" 42 bytes [built] [code generated]
28 modules
webpack 5.24.3 compiled successfully in 3282 ms
```

## No `babel-preset`

It turns out, Node 10 already supports everything so `babel-preset` can just be dropped.

## Without others

Look at this beauty, with -40% build time 

```js 
asset addons-linter.js 648 KiB [emitted] (name: addons-linter) 1 related asset
asset rules/javascript/webextension-deprecated-api.js 497 KiB [compared for emit] (name: rules/javascript/webextension-deprecated-api) 1 related asset
asset rules/javascript/webextension-unsupported-api.js 459 KiB [compared for emit] (name: rules/javascript/webextension-unsupported-api) 1 related asset
asset rules/javascript/webextension-api-compat-android.js 459 KiB [compared for emit] (name: rules/javascript/webextension-api-compat-android) 1 related asset
asset rules/javascript/webextension-api-compat.js 459 KiB [compared for emit] (name: rules/javascript/webextension-api-compat) 1 related asset
asset rules/javascript/webextension-api.js 450 KiB [compared for emit] (name: rules/javascript/webextension-api) 1 related asset
asset rules/javascript/deprecated-entities.js 73.8 KiB [compared for emit] (name: rules/javascript/deprecated-entities) 1 related asset
asset rules/javascript/global-require-arg.js 73.4 KiB [compared for emit] (name: rules/javascript/global-require-arg) 1 related asset
asset rules/javascript/opendialog-remote-uri.js 73.3 KiB [compared for emit] (name: rules/javascript/opendialog-remote-uri) 1 related asset
asset rules/javascript/opendialog-nonlit-uri.js 73.2 KiB [compared for emit] (name: rules/javascript/opendialog-nonlit-uri) 1 related asset
asset rules/javascript/content-scripts-file-absent.js 39.5 KiB [compared for emit] (name: rules/javascript/content-scripts-file-absent) 1 related asset
asset rules/javascript/event-listener-fourth.js 37 KiB [compared for emit] (name: rules/javascript/event-listener-fourth) 1 related asset
runtime modules 793 bytes 4 modules
modules by path ./src/ 647 KiB
  modules by path ./src/schema/ 445 KiB 64 modules
  modules by path ./src/rules/ 20.8 KiB 16 modules
  modules by path ./src/*.js 68.9 KiB 11 modules
  modules by path ./src/messages/*.js 45.4 KiB 9 modules
  modules by path ./src/scanners/*.js 24.1 KiB 8 modules
  modules by path ./src/parsers/*.js 43 KiB 6 modules
  ./src/blocked_content_script_hosts.txt 45 bytes [built] [code generated]
modules by path external "addons-scanner-utils/dist/ 84 bytes
  external "addons-scanner-utils/dist/io/utils" 42 bytes [built] [code generated]
  external "addons-scanner-utils/dist/io" 42 bytes [built] [code generated]
28 modules
webpack 5.24.3 compiled successfully in 2852 ms
```

</details>

### CODE REVIEW

**Comment by willdurand** [comment]
> Why is there babel-env?

It was probably needed in the past.

> Does this package run in the browser?

No.

> I ran some tests and here are the results

Looks good to me, I am fine removing build related packages if they are useless.

**Comment by fregante** [comment]
It worked locally, but not on CI. I'll re-add what's necessary.

**Comment by fregante** [comment]
> > Why is there babel-env?
> 
> It was probably needed in the past.

I was testing it on Node 15 locally and everything worked without it, but Node 10 doesn't have `?.` for example. I had to restore it. That's probably one of the few applicable transforms since size was similar.

Here's the current build time:

```js 

asset addons-linter.js 648 KiB [compared for emit] (name: addons-linter) 1 related asset
asset rules/javascript/webextension-deprecated-api.js 497 KiB [compared for emit] (name: rules/javascript/webextension-deprecated-api) 1 related asset
asset rules/javascript/webextension-unsupported-api.js 459 KiB [compared for emit] (name: rules/javascript/webextension-unsupported-api) 1 related asset
asset rules/javascript/webextension-api-compat-android.js 459 KiB [compared for emit] (name: rules/javascript/webextension-api-compat-android) 1 related asset
asset rules/javascript/webextension-api-compat.js 459 KiB [compared for emit] (name: rules/javascript/webextension-api-compat) 1 related asset
asset rules/javascript/webextension-api.js 450 KiB [compared for emit] (name: rules/javascript/webextension-api) 1 related asset
asset rules/javascript/deprecated-entities.js 73.8 KiB [compared for emit] (name: rules/javascript/deprecated-entities) 1 related asset
asset rules/javascript/global-require-arg.js 73.4 KiB [compared for emit] (name: rules/javascript/global-require-arg) 1 related asset
asset rules/javascript/opendialog-remote-uri.js 73.3 KiB [compared for emit] (name: rules/javascript/opendialog-remote-uri) 1 related asset
asset rules/javascript/opendialog-nonlit-uri.js 73.2 KiB [compared for emit] (name: rules/javascript/opendialog-nonlit-uri) 1 related asset
asset rules/javascript/content-scripts-file-absent.js 39.5 KiB [compared for emit] (name: rules/javascript/content-scripts-file-absent) 1 related asset
asset rules/javascript/event-listener-fourth.js 37 KiB [compared for emit] (name: rules/javascript/event-listener-fourth) 1 related asset
runtime modules 793 bytes 4 modules
modules by path ./src/ 647 KiB
  modules by path ./src/schema/ 445 KiB 64 modules
  modules by path ./src/rules/ 20.8 KiB 16 modules
  modules by path ./src/*.js 68.9 KiB 11 modules
  modules by path ./src/messages/*.js 45.4 KiB 9 modules
  modules by path ./src/scanners/*.js 24.1 KiB 8 modules
  modules by path ./src/parsers/*.js 43 KiB 6 modules
  ./src/blocked_content_script_hosts.txt 45 bytes [built] [code generated]
modules by path external "addons-scanner-utils/dist/ 84 bytes
  external "addons-scanner-utils/dist/io/utils" 42 bytes [built] [code generated]
  external "addons-scanner-utils/dist/io" 42 bytes [built] [code generated]
28 modules
webpack 5.24.3 compiled successfully in 3399 ms
```

**Comment by willdurand** [comment]
> I was testing it on Node 15 locally and everything worked without it, but Node 10 doesn't have `?.` for example. I had to restore it. That's probably one of the few applicable transforms since size was similar.

We're going to drop Node 10 support very soon. Let's keep the transform for now and create an issue to remove it when we officially drop Node 10 support.

**Comment by fregante** [comment]
This should be good to go. Any further changes can be decided once the minimum version is changed. Preset-env might still be useful then, it just depends on whoever makes that change.

File: package.json [hunk 1]
```diff
@@ -57,6 +59,7 @@
     "espree": "7.3.1",
     "esprima": "4.0.1",
     "fluent-syntax": "0.13.0",
+    "fsevents": "2.3.2",
```
Comment (willdurand) [inline package.json:62]: We don't need this dependency. NPM adds and removes it automatically.. and it's annoying. Could you remove it please?

**Review by willdurand — COMMENTED** [review]
Looks good to me (once `fsevents` is removed). @rpl do you want to take a quick look?

File: package.json (hunk 1 above)
Comment (fregante) [reply package.json:62]: It actually appears in `optionalDependencies` as well. If I run `npm rm fsevents` it's removed from both. I sent a commit with this change

**Review by rpl — APPROVED** [review]
lgtm too (and it does also look like a more than reasonable cleanup, thanks!)

**Pull request merged**